### PR TITLE
Support for a missing "Jinxed Juju" unique amulet mod

### PR DESCRIPTION
**Pull request by mullinger**
Previously, the special mod the the Jinxed Juju Amulet was not supported:

> 10% of Damage from Hits is taken from your Spectres' Life before you

This commit adds support for the mod in regards to the damage taken calculations.

grafik

I assume that the calculations in-game might be slightly different, though the differences are probably in the sub-percent range.
This patch should give a pretty good addition for now.

### CODE REVIEW

**Comment by Regisle** [comment]
This needs to only be active if your specters have enough life to take the damage (make a config option) as there are builds who use the amulet without specters

**Comment by mullinger** [comment]
@Regisle Very good point, thanks for your comment.
Instead of adding yet another config options I thought about checking if the player actually has an active spectre skill.

Without active spectre skill, or of the spectre skill is disabled the mod from the amulet will not be applied.

What do you think?

**Comment by ppoelzl** [comment]
There should be some kind of calculation on how much damage is actually transferable to the spectre.
Also in your code you spelled it "specter", not "spectre", which is our preferred spelling @mullinger

**Comment by Wires77** [comment]
> 
> 
> There should be some kind of calculation on how much damage is actually transferable to the spectre.

Eh, I'm not sure this is necessary, even the weakest spectres have almost 10k health, so you'd need to take 100k damage before that's something to worry about.

I'm all for not adding more config options, too, so just basing it off of the player having an active spectre skill seems like the right way to go to me.

**Comment by mullinger** [comment]
E.g. Apes have unbuffed 16k max HP without **any** investment .
At lvl 20 gem you can have 3 apes with a total pool of around 48k HP.

I'm down with any decision (Config option or without) as I can see the benefits of both implementations. Just need a decision :-)

**Comment by deathbeam** [comment]
This is not 10% less damage taken, this should work like mind over matter because its same mechanic.

File: Modules/CalcPerform.lua [hunk 1]
```diff
@@ -425,6 +425,19 @@ local function doActorMisc(env, actor)
 			end
 			modDB.multipliers["BuffOnSelf"] = (modDB.multipliers["BuffOnSelf"] or 0) + 1
 		end
+		if modDB:Sum("BASE", nil, "DamageFromHitsTakenFromSpectres") > 0 then
+			local isActiveSpectre = false
+			for _, activeSkill in ipairs(env.player.activeSkillList) do
+				if activeSkill.skillFlags.spectre then
+					isActiveSpectre = true
+				end
+			end
```
Comment (Wires77) [inline Modules/CalcPerform.lua:434]: This loop is already done on https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/3e28ad5599ebcddf8377cb3cbb433fefe4fb7ff4/Modules/CalcPerform.lua#L767-L770

~~Can you change this code to use that variable, instead?~~

Actually, on second thought, you can do this entirely in ModParser.lua, see that for a suggestion

File: Modules/ModParser.lua [hunk 2]
```diff
@@ -258,6 +258,7 @@ local modNameList = {
 	["recovery rate of life and energy shield"] = { "LifeRecoveryRate", "EnergyShieldRecoveryRate" },
 	["maximum life, mana and global energy shield"] = { "Life", "Mana", "EnergyShield", tag = { type = "Global" } },
 	["non-chaos damage taken bypasses energy shield"] = {"PhysicalEnergyShieldBypass", "LightningEnergyShieldBypass", "ColdEnergyShieldBypass", "FireEnergyShieldBypass" },
+	["damage from hits is taken from your spectres' life before you"] = "DamageFromHitsTakenFromSpectres",
```
Comment (Wires77) [inline Modules/ModParser.lua:261]: I haven't tested this, but see if this works instead

```suggestion
	["damage from hits is taken from your spectres' life before you"] = { "DamageTakenWhenHit", tag = { type = "SkillName", skillName = "Raise Spectre" } },
```

**Review by Wires77 — CHANGES_REQUESTED** [review]
If we're fine assuming the spectres will always have enough health to give you the damage mitigation, I've added a suggestion for a simpler way to implement this

**Comment by Wires77** [comment]
@deathbeam is correct, this implementation isn't right.  It should act as another multiplier on top of other reduced * less damage modifiers you have.

I can merge this as a 10% less damage taken, and open an issue to fix it properly later, or we can keep it open and do it properly from the start.  React if you don't want to comment but want to vote on one over the other.

✔️ Merge now, fix later
❌ Don't merge until properly implemented

**Comment by ppoelzl** [comment]
No need to rush out unfinished features "just because".

**Comment by gbhwnpbshb** [comment]
I'm a bit confused

**Comment by QuickStick123** [comment]
Superseded by https://github.com/PathOfBuildingCommunity/PathOfBuilding/pull/5943

**Pull request closed without merging**